packages/l1-l3-teleport/scripts/initiate-deposit.ts

```diff
@@ -0,0 +1,151 @@
+   * B) Manually redeem the retryable on L3 to pay the fee (this can be done by anyone, not just the sender of the deposit). 
+   *    If the fee token is unavailable on L1 or skipFeeToken is set to true, manual redemption is required.
+   * 
+   * If the L3 uses ETH for fees, all retryables are paid for when initiating the deposit by default and no additional steps are required.
+   */
+  const l1ChainId = (await l1Provider.getNetwork()).chainId
+  const l2ChainId = (await l2Provider.getNetwork()).chainId
+  const l3ChainId = (await l3Provider.getNetwork()).chainId
+  const signerAddress = await l1Signer.getAddress()
+  const l3FeeTokenL1Address = await bridger.l1FeeTokenAddress(l1Provider, l2Provider)
+  const feeTokenSymbol = bridger.l2FeeTokenAddress ? await ERC20__factory.connect(bridger.l2FeeTokenAddress, l2Provider).symbol() : 'ETH'
+  console.log('L1 Token:', params.l1Token)
+  console.log('L3 Fee Token:', feeTokenSymbol || 'ETH')
+  console.log('L3 Fee Token Address on L2:', bridger.l2FeeTokenAddress || ethers.constants.AddressZero)
+  console.log('L3 Fee Token Address on L1:', l3FeeTokenL1Address || ethers.constants.AddressZero)
+  console.log('Recipient:', params.l3Recipient || signerAddress)
+  console.log('Amount:', params.amount.toString())
+  console.log('L1 chain id:', l1ChainId)
+  console.log('L2 chain id:', l2ChainId)
+  console.log('L3 chain id:', l3ChainId)
+
+  /**
+   * Get a deposit request
+   */
+  console.log('Getting deposit request...')
+  const depositRequest = await bridger.getDepositRequest({
+    l1Signer,
+    erc20L1Address: params.l1Token,
+    amount: params.amount,
+    l2Provider,
+    l3Provider,
+    to: params.l3Recipient, // optional, defaults to signer's address
+    /**
+     * Optional, defaults to false.
+     * Skip paying the L2->L3 fee if the L3 doesn't use ETH for fees
+     * This has no effect if the L3 uses ETH for fees.
+     */
+    skipFeeToken: params.skipFeeToken
+  })
+  console.log('Done')
+
+  /**
+   * If the deposit request has nonzero feeTokenAmount, we are paying for the L2->L3 retryable when initiating the deposit.
+   * We must approve the fee token on L1.
+   */
+  if (depositRequest.feeTokenAmount.gt(0)) {
+    console.log('Approving fee token on L1...')
+    await (await bridger.approveFeeToken({
+      l1Signer,
+      l2Provider,
+      amount: depositRequest.feeTokenAmount,
+    })).wait()
+    console.log('Done')
+  }
+
+  /**
+   * Approve the token on L1
+   */
+  console.log('Approving token on L1...')
+  await (await bridger.approveToken({
+    erc20L1Address: params.l1Token,
+    l1Signer,
+    amount: params.amount // optional, defaults to max
+  })).wait()
+  console.log('Done')
+
+  /**
+   * Deposit the token
+   */
+  console.log('Depositing token...')
+  const depositTx = await bridger.deposit({
+    txRequest: depositRequest.txRequest,
+    l1Signer
+  })
+  await depositTx.wait()
+  console.log('Done')
+
+  console.log('Initiated deposit:', depositTx.hash)
```

Comment:
It'd be nice to explain what happens after the deposit is initiated, and also add the command to monitor the transaction just below `yarn monitor-deposit-status ${depositTx.hash}`, so users know that they have to wait for the deposits to go through and that they can use that command to monitor.

(The same applies for initiate-eth-deposit)

Reply:
something like this?
03bec8b778477f46c7475e935b6f1b8a9d68284b

Reply:
This is great, thanks!